tiledb/api/src/stats.rs

```diff
@@ -0,0 +1,85 @@
+use std::collections::HashMap;
+
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+
+use crate::error::Error;
+use crate::Result as TileDBResult;
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct Metrics {
+    pub timers: HashMap<String, f64>,
+    pub counters: HashMap<String, u64>,
+}
+
+pub fn enable() -> TileDBResult<()> {
+    let c_ret = unsafe { ffi::tiledb_stats_enable() };
+
+    if c_ret == ffi::TILEDB_OK {
+        Ok(())
+    } else {
+        Err(Error::LibTileDB(String::from("Failed to enable stats.")))
+    }
+}
+
+pub fn disable() -> TileDBResult<()> {
+    let c_ret = unsafe { ffi::tiledb_stats_disable() };
+
+    if c_ret == ffi::TILEDB_OK {
+        Ok(())
+    } else {
+        Err(Error::LibTileDB(String::from("Failed to disable stats.")))
+    }
+}
+
+pub fn reset() -> TileDBResult<()> {
+    let c_ret = unsafe { ffi::tiledb_stats_reset() };
+
+    if c_ret == ffi::TILEDB_OK {
+        Ok(())
+    } else {
+        Err(Error::LibTileDB(String::from("Failed to reset stats.")))
+    }
+}
+
+pub fn dump() -> TileDBResult<Option<String>> {
+    let mut c_str = std::ptr::null_mut::<std::ffi::c_char>();
+    let c_ret = unsafe {
+        ffi::tiledb_stats_dump_str(&mut c_str as *mut *mut std::ffi::c_char)
+    };
+    if c_ret == ffi::TILEDB_OK {
+        assert!(!c_str.is_null());
+        let stats_dump = unsafe { std::ffi::CStr::from_ptr(c_str) };
+        let stats_dump_rust_str = stats_dump.to_string_lossy().into_owned();
+        let free_c_ret = unsafe {
+            ffi::tiledb_stats_free_str(&mut c_str as *mut *mut std::ffi::c_char)
```

Comment:
The only thing I'm not 100% certain about. Might be because I've still got C++ brain, but can we guarantee that we won't panic inbetween these lines and leak that pointer? Usually panic means end of the process, but apparently not always, so we could technically end up leaking the stats string if someone using tiledb-rs is catching panics. I think?

Reply:
There is apparently an "unwinding panic" which calls `drop` of things on the stack in the manner you would expect from a C++ thrown exception.  I wouldn't really expect that from `to_string_lossy()` unless maybe it runs out of memory. The doc makes no mention of a panic scenario.

I'd never say no to a struct that wrapped this thing and did teh `tiledb_stats_free_str` in a `drop` implementation though.

Reply:
I’m fine leaving it as a follow up since it seems very unlikely to be an actual issue in practice. I’ll open an issue to remind me to file the story since I’m on mobile. Good to merge by me.